File: backend/app/schemas/file.py
"""File operation schemas."""

from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class FileNode(BaseModel):
    name: str
    type: str
    path: str
    size: Optional[int] = None
    modified: Optional[str] = None
    children: Optional[list["FileNode"]] = None


class FileTreeResponse(BaseModel):
    project_id: UUID
    root: FileNode


class FileContentResponse(BaseModel):
    path: str
    name: str = ""
    content: str
    encoding: str = "utf-8"
    size: int
    modified: str = ""
    is_binary: bool = False


class GitStatusResponse(BaseModel):
    project_id: UUID
    is_git_repo: bool = True
    current_branch: Optional[str] = None
    modified_files: list[str] = []
    staged_files: list[str] = []
    untracked_files: list[str] = []
    ahead: int = 0
    behind: int = 0

File: frontend/src/types/file.ts
// File system-related types

export interface FileNode {
  name: string
  type: "file" | "folder"
  path: string
  size?: number
  modified?: string
  children?: FileNode[]
}

export interface FileTreeResponse {
  project_id: string
  project_path: string
  tree: FileNode[]
  root: FileNode
}

export interface FileContentResponse {
  path: string
  name: string
  content: string
  size: number
  encoding: string
  modified?: string
  is_binary?: boolean
}

export interface GitStatusResponse {
  project_id: string
  is_git_repo: boolean
  current_branch: string | null
  modified_files: string[]
  staged_files: string[]
  untracked_files: string[]
  ahead: number
  behind: number
}
